### PR TITLE
Feature/summary audit#705

### PR DESCRIPTION
**Pull request by LindsayYoung**
Making sure we have all the variables for the 3 kinds of reports. 

I am going to make partials for each and make them dependent on the committee type so they show the right way on the committee profile.

### CODE REVIEW

**Comment by LindsayYoung** [comment]
I am deviating form the FEC layout of the summaries in one way-

I am putting net contributions under total contributions and net operating expenses under total operating expenses. That will give us a clear divide of things that are not totaled versus things that are and I am putting like things together. The change does make the layout in a different order from the underlying form. Feel free to change that back if you don't like it.

**Comment by LindsayYoung** [comment]
There is one more cleanly-shaven yack for the project.

File: templates/partials/committee-report-house-senate.html [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{% with committee=context() %}
+
+{% set table_data = [
+     (reports.0.individual_itemized_contributions_period, 'Itemized Individual Contributions'),
+     (reports.0.individual_unitemized_contributions_period, 'Unitemized Individual Contributions'),
+     (reports.0.total_individual_contributions_period, 'Total Individual Contributions'),
+     (reports.0.political_party_committee_contributions_period,'Party Committee Contributions'),
+     (reports.0.other_political_committee_contributions_period,'Other Committee Contributions'),
+     (reports.0.candidate_contribution_period, 'Candidate Contributions'),
+     (reports.0.total_contributions_period, 'Total Contributions'),
```
Comment (noahmanger) [inline templates/partials/committee-report-house-senate.html]: I thought we weren't including sub-totals like Total Contributions

File: templates/partials/committee-report-house-senate.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+     (reports.0.transfers_from_other_authorized_committee_period, 'Transfer from Authorized Committees'),
+     (reports.0.loans_made_by_candidate_period, 'Loans Made by Candidate'),
+     (reports.0.all_other_loans, 'Other Loans'),
+     (reports.0.all_other_loans_period, 'Total Loans Received'),
```
Comment (noahmanger) [inline templates/partials/committee-report-house-senate.html]: And this one

File: templates/partials/committee-report-house-senate.html (hunk 1 above)
Comment (LindsayYoung) [inline templates/partials/committee-report-house-senate.html]: We went back and forth on it, I thought we ended it up saying we were just going to show everything. We can take them out if you want.

File: templates/partials/committee-report-house-senate.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+     (reports.0.individual_itemized_contributions_period, 'Itemized Individual Contributions'),
+     (reports.0.individual_unitemized_contributions_period, 'Unitemized Individual Contributions'),
+     (reports.0.total_individual_contributions_period, 'Total Individual Contributions'),
+     (reports.0.political_party_committee_contributions_period,'Party Committee Contributions'),
+     (reports.0.other_political_committee_contributions_period,'Other Committee Contributions'),
+     (reports.0.candidate_contribution_period, 'Candidate Contributions'),
+     (reports.0.total_contributions_period, 'Total Contributions'),
+     (reports.0.transfers_from_other_authorized_committee_period, 'Transfer from Authorized Committees'),
+     (reports.0.loans_made_by_candidate_period, 'Loans Made by Candidate'),
+     (reports.0.all_other_loans, 'Other Loans'),
```
Comment (noahmanger) [inline templates/partials/committee-report-house-senate.html]: This is wrong

File: templates/partials/committee-totals-house-senate.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+     (totals.0.individual_contributions, 'Total Individual Contributions'),
+     (totals.0.political_party_committee_contributions,'Party Committee Contributions'),
+     (totals.0.other_political_committee_contributions,'Other Committee Contributions'),
+     (totals.0.candidate_contribution, 'Candidate Contributions'),
+     (totals.0.contributions, 'Total Contributions'),
+     (totals.0.net_contributions, 'Net Contributions'),
+     (totals.0.transfers_from_other_authorized_committee, 'Transfer from Authorized Committees'),
+     (totals.0.loans_made_by_candidate, 'Loans Made by Candidate'),
+     (totals.0.all_other_loans, 'Other Loans'),
+     (totals.0.all_loans_received, 'Total Loans Received'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-house-senate.html]: This is showing an error

File: templates/partials/committee-totals-house-senate.html [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+{{ totals_table('Total receipts', totals.0.receipts, table_data,
+    committee.report_year, header_description='Money received by the committee' ) }}
+
+{% set table_data = [
+     (totals.0.operating_expenditures, 'Operating expenditures'),
+     (totals.0.net_operating_expenditures, 'Operating Expenditures'),
+     (totals.0.transfers_to_affiliated_committee, 'Transfers to authorized committees'),
+     (totals.0.loan_repayments_candidate_loans, 'Candidate Loan Repayments'),
+     (totals.0.loan_repayments_other_loans, 'Other Loan Repayments'),
+     (totals.0.loan_repayments_made, 'Total Loan Repayments'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-house-senate.html]: Error

File: templates/partials/committee-totals-house-senate.html [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+     (totals.0.other_receipts, 'Other Receipts'),
+   ]
+%}
+{{ totals_table('Total receipts', totals.0.receipts, table_data,
+    committee.report_year, header_description='Money received by the committee' ) }}
+
+{% set table_data = [
+     (totals.0.operating_expenditures, 'Operating expenditures'),
+     (totals.0.net_operating_expenditures, 'Operating Expenditures'),
+     (totals.0.transfers_to_affiliated_committee, 'Transfers to authorized committees'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-house-senate.html]: Error

File: templates/partials/committee-report-pac-party.html [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+     (reports.0.transfers_from_nonfed_account_period, 'Total Non-Federal Transfers'),
+     (reports.0.total_fed_receipts_period, 'Total Federal Receipts'),
+   ]
+%}
+{{ totals_table('Total receipts', reports.0.total_receipts_period, table_data,
+    committee.report_year, header_description='Money received by the committee' ) }}
+
+{% set table_data = [
+     (reports.0.shared_fed_operating_expenditures_period, 'Federal Allocated Operating Expenditures'),
+     (reports.0.shared_nonfed_operating_expenditures_period, 'Non-Federal Operating Expenditures'),
```
Comment (noahmanger) [inline templates/partials/committee-report-pac-party.html]: Error

File: templates/partials/committee-report-pac-party.html [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+     (reports.0.total_fed_disbursements_period, 'Total Federal Disbursements'),
+   ]
+%}
+{{ totals_table('Total Disbursements', reports.0.total_disbursements_period, table_data,
+    committee.report_year, header_description='Money spent by the committee' ) }}
+
+{{ totals_table('Beginning cash on hand', reports.0.cash_on_hand_beginning_period) }}
+{{ totals_table('Ending cash on hand', reports.0.cash_on_hand_end_period) }}
+
+{{ totals_table('Net Contributions', reports.0.net_contributions) }}
```
Comment (noahmanger) [inline templates/partials/committee-report-pac-party.html]: Error

File: templates/partials/committee-report-presidential.html [hunk 9]
```diff
@@ -0,0 +1,50 @@
+{% with committee=context() %}
+{% set table_data = [
+     (reports.0.federal_funds_period,'Federal Funds'),
+     (reports.0.individual_itemized_contributions, 'Itemized Individual Contributions'),
```
Comment (noahmanger) [inline templates/partials/committee-report-presidential.html]: Error

File: templates/partials/committee-report-presidential.html [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+     (reports.0.individual_unitemized_contributions, 'Unitemized Individual Contributions'),
```
Comment (noahmanger) [inline templates/partials/committee-report-presidential.html]: Error

File: templates/partials/committee-report-presidential.html [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+{{ totals_table('Total receipts', reports.0.total_receipts_period, table_data,
+    committee.report_year, header_description='Money received by the committee' ) }}
+
+{% set table_data = [
+     (reports.0.operating_expenditures_period, 'Operating expenditures'),
+     (reports.0.transfer_to_other_authorized_committee_period, 'Transfers to Authorized Committees'),
+     (reports.0.repayments_loans_made_by_candidate_period, 'Candidate Loan Repayments'),
+     (reports.0.repayments_other_loans_period, 'Other Loan Repayments'),
+     (reports.0.total_loan_repayments_made_period, 'Total Loan Repayments'),
+     (reports.0.refunded_individual_contributions_period, 'Individual Refunds'),
```
Comment (noahmanger) [inline templates/partials/committee-report-presidential.html]: Error

File: templates/partials/committee-totals-presidential.html [hunk 12]
```diff
@@ -0,0 +1,48 @@
+{% with committee=context() %}
+{% set table_data = [
+     (totals.0.federal_funds,'Federal Funds'),
+     (totals.0.individual_itemized_contributions, 'Itemized Individual Contributions'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-presidential.html]: Error

File: templates/partials/committee-totals-presidential.html [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+     (totals.0.individual_unitemized_contributions, 'Unitemized Individual Contributions'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-presidential.html]: Error

File: templates/partials/committee-totals-presidential.html [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+     (totals.0.individual_contributions, 'Total Individual Contributions'),
+     (totals.0.political_party_committee_contributions,'Party Committee Contributions'),
+     (totals.0.other_political_committee_contributions,'Other Committee Contributions'),
+     (totals.0.candidate_contribution, 'Candidate Contributions'),
+     (totals.0.contributions, 'Total Contributions'),
+     (totals.0.transfers_from_affiliated_committee, 'Transfer from Affiliated Committees'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-presidential.html]: Error

File: templates/partials/committee-totals-presidential.html [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+     (totals.0.loans_received_from_candidate, 'Loans Made by Candidate'),
+     (totals.0.other_loans_received, 'Other Loans'),
+     (totals.0.all_loans_received, 'Total Loans Received'),
+     (totals.0.offsets_to_operating_expenditures, 'Offsets to Operating Expenditures'),
+     (totals.0.offsets_to_fundraising_expenditures, 'Fundraising Offsets'),
+     (totals.0.offsets_to_legal_accounting, 'Legal and Accounting Offsets'),
+     (totals.0.total_offsets_to_operating_expenditures, 'Total Offsets'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-presidential.html]: Error

File: templates/partials/committee-totals-presidential.html [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+     (totals.0.other_receipts, 'Other Receipts'),
+   ]
+%}
+{{ totals_table('Total receipts', totals.0.receipts, table_data,
+    committee.report_year, header_description='Money received by the committee' ) }}
+
+{% set table_data = [
+     (totals.0.operating_expenditures, 'Operating expenditures'),
+     (totals.0.transfers_to_other_authorized_committee, 'Transfers to Authorized Committees'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-presidential.html]: Error

File: templates/partials/committee-totals-presidential.html [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+     (totals.0.repayments_loans_made_by_candidate, 'Candidate Loan Repayments'),
+     (totals.0.repayments_other_loans, 'Other Loan Repayments'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-presidential.html]: Error

File: templates/partials/committee-totals-pac-party.html [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+%}
+{{ totals_table('Total receipts', totals.0.receipts, table_data,
+    committee.report_year, header_description='Money received by the committee' ) }}
+
+{% set table_data = [
+     (totals.0.shared_fed_operating_expenditures, 'Federal Allocated Operating Expenditures'),
+     (totals.0.shared_nonfed_operating_expenditures, 'Non-Federal Operating Expenditures'),
+     (totals.0.operating_expenditures, 'Total Operating Expenditures'),
+     (totals.0.net_operating_expenditures, 'Net Operating Expenditures'),
+     (totals.0.transfers_to_other_authorized_committee, 'Transfers to Authorized Committees'),
```
Comment (noahmanger) [inline templates/partials/committee-totals-pac-party.html]: Error

**Comment by noahmanger** [comment]
Ok, I think I got them all.

File: templates/partials/committee-report-house-senate.html (hunk 1 above)
Comment (noahmanger) [inline templates/partials/committee-report-house-senate.html]: Oh sorry. Yeah I think we should leave them out becuase there's currently
no way of showing them as a cumulative total of the other lines.

On Wed, May 13, 2015 at 4:20 PM, Lindsay Young <notifications@github.com>
wrote:

> In templates/partials/committee-report-house-senate.html
> <https://github.com/18F/openFEC-web-app/pull/158#discussion_r30285849>:
>
> > @@ -0,0 +1,46 @@
> > +{% with committee=context() %}
> > +
> > +{% set table_data = [
> > +     (reports.0.individual_itemized_contributions_period, 'Itemized Individual Contributions'),
> > +     (reports.0.individual_unitemized_contributions_period, 'Unitemized Individual Contributions'),
> > +     (reports.0.total_individual_contributions_period, 'Total Individual Contributions'),
> > +     (reports.0.political_party_committee_contributions_period,'Party Committee Contributions'),
> > +     (reports.0.other_political_committee_contributions_period,'Other Committee Contributions'),
> > +     (reports.0.candidate_contribution_period, 'Candidate Contributions'),
> > +     (reports.0.total_contributions_period, 'Total Contributions'),
>
> We went back and forth on it, I thought we ended it up saying we were just
> going to show everything. We can take them out if you want.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/18F/openFEC-web-app/pull/158/files#r30285849>.
>



-- 
Noah Manger
18F <http://18f.gsa.gov> | Work: 415-696-6146

**Comment by LindsayYoung** [comment]
Thank you so much @noahmanger I will get on those now.

**Comment by noahmanger** [comment]
LGTM :+1:

**Pull request merged**